### PR TITLE
chore: remove option identifierPattern

### PR DESCRIPTION
**Pull request by 3cp**
This option added a "pattern" flag to Identifer node, but this feature was rejected by ESTree, https://github.com/estree/estree/issues/196 KFlash prematurelly added this when reading it.
Since it's not in ESTree spec, nobody would never relied on it. In this sense, not a breaking change.

### CODE REVIEW

**Review by fisker — APPROVED** [review]
Make sense!

**Comment by fisker** [comment]
> At least 1 approving review is required by reviewers with write access.

I don't have write access, so my approval won't unblock.

**Comment by 3cp** [comment]
> > At least 1 approving review is required by reviewers with write access.
> 
> I don't have write access, so my approval won't unblock.

Thx! That confused hell out of me. Write permission assigned.

**Comment by fisker** [comment]
> Write permission assigned.

I don't know that I'm a member here..

> That confused hell out of me. 

The check mark was gray (not green) before

**Pull request merged**
